### ISSUE TITLE
Using validate with non-input elements

### ISSUE THREAD

**Issue by andrewchch** [jquery-validation/jquery-validation#484]
Hi there,

I see that the plugin is restricted to work with only input elements, but it would be very handy for me to be able to validate non-input elements as well. Specifically, I have UL that is part of a compound control (a multiselect) that I would like to validate as having any list items. At the moment, this is not possible with this plugin, but is there any particular reason why it couldn't be modified accordingly?

Cheers, Andrew.

**Comment by afilina** [comment]
+1, we really need this.

I'm fixing legacy (and just plain crap) code for a living and this plugin is making my life so much harder by not allowing random elements to be validated. I have to do this hack over and over again, but always custom-tailored. SO is also full of people crying for help over this.

**Comment by staabm** [comment]
@afilina how does a use-case look like in which you validate a non-input field? what do you validate at all on the clientside in this case? how does the user input data in your "element"?

**Comment by afilina** [comment]
@staabm Developers write a bunch of convoluted "plugins" for fancy behaviour. For example, the one I battled with tonight was listening to clicks on list elements in order to generate spans for them and then inserting those new spans into a whole other element. Terrible, I know.

I eventually had to listen to those clicks too and then counting how many spans it inserted, just to get this number into a hidden field. Only then I was able to validate using this plugin. It would have been less crappy to just validate the container with its spans directly, because then it would have been possible to generalize the solution instead of copy-pasting in a gazillion files.

Sometimes the code I have to deal with it so bad that my only chance at sanity is to create reusable solutions, even if they're not pretty. That project has almost a million lines of code, so it's hard enough as it is to fix.
